lib/src/git.rs

```diff
@@ -2555,23 +2593,10 @@ fn subprocess_push_refs(
         .map(|full_refspec| RefToPush::new(full_refspec, qualified_remote_refs_expected_locations))
         .collect();
 
-    let push_stats = git_ctx.spawn_push(remote_name, &refs_to_push, &mut callbacks)?;
-    tracing::debug!(?push_stats);
-
-    if !push_stats.rejected.is_empty() {
-        let mut refs_in_unexpected_locations = push_stats.rejected;
-        refs_in_unexpected_locations.sort();
-        Err(GitPushError::RefInUnexpectedLocation(
-            refs_in_unexpected_locations,
-        ))
-    } else if !push_stats.remote_rejected.is_empty() {
-        let mut rejected_refs = push_stats.remote_rejected;
-        rejected_refs.sort();
-        // remote rejected refs because of remote config (e.g., no push on main)
-        Err(GitPushError::RefUpdateRejected(rejected_refs))
-    } else {
-        Ok(())
-    }
+    let mut push_stats = git_ctx.spawn_push(remote_name, &refs_to_push, &mut callbacks)?;
+    push_stats.rejected.sort();
+    push_stats.remote_rejected.sort();
```

Comment:
nit: sort `push_stats.pushed` too? (or these lists might be sorted by `git` CLI, but I'm not sure.)